specs/core/1_shard-data-chains.md

```diff
@@ -75,207 +91,191 @@ The following types are defined, mapping into `DomainType` (little endian):
 
 | Name | Value |
 | - | - |
-| `PLACEHOLDER` | `2**32` |
+| `PLACEHOLDER` | `2**3` |
 
 ## Data structures
 
-### `ShardBlockBody`
+_Note: the shard block header structure is carefully designed so that all of the values have the same depth in a hash tree implementation, so `hash_tree_root(SSZ_partial(x)) == hash_tree_root(x)` (using the "left-to-right leaves" scheme [here](https://github.com/ethereum/eth2.0-specs/issues/1303)), which allows shard block headers to look like an SSZ object when in the crosslink structure. This is done by balancing it so that 7 or 8 items are on the left side (the "core") and two 96-byte (ie. 3*2 = 6 chunk) items are on the right side. Change with care._
+
+### `ShardBlockHeader`
+
+```python
+class ShardBlockHeader(Container):
+    core: ShardBlockCore
+    signatures: ShardBlockSignatures
+```
+
+### `ShardBlock`
 
 ```python
-class ShardBlockBody(Container):
-    data: Vector[Bytes[PLACEHOLDER], BYTES_PER_SHARD_BLOCK_BODY]
+class ShardBlock(Container):
+    core: ExtendedShardBlockCore
+    signatures: ShardBlockSignatures
 ```
 
-### `ShardAttestation`
+### `ShardBlockSignatures`
 
 ```python
-class ShardAttestation(Container):
-    class data(Container):
-        slot: Slot
-        shard: Shard
-        shard_block_root: Hash
-    aggregation_bits: Bitlist[PLACEHOLDER]
-    aggregate_signature: BLSSignature
+class ShardBlockSignatures(Container):
+    attestation_signature: BLSSignature
+    proposer_signature: BLSSignature
 ```
 
-### `ShardBlock`
+### `ShardBlockCore`
 
 ```python
-class ShardBlock(Container):
-    slot: Slot
-    shard: Shard
+    slot: ShardSlot
     beacon_chain_root: Hash
     parent_root: Hash
-    data: ShardBlockBody
+    data_root: Hash
     state_root: Hash
-    attestations: List[ShardAttestation, PLACEHOLDER]
-    signature: BLSSignature
+    total_bytes: uint64
+    attester_bitfield: Bitvector[MAX_PERSISTENT_COMMITTEE_SIZE * 2]
 ```
 
-### `ShardBlockHeader`
+### `ExtendedShardBlockCore`
 
 ```python
-class ShardBlockHeader(Container):
-    slot: Slot
-    shard: Shard
+    slot: ShardSlot
     beacon_chain_root: Hash
     parent_root: Hash
-    body_root: Hash
+    data_root: Bytes[SHARD_BLOCK_SIZE_LIMIT - SHARD_HEADER_SIZE]
     state_root: Hash
-    attestations: List[ShardAttestation, PLACEHOLDER]
-    signature: BLSSignature
+    total_bytes: uint64
+    attester_bitfield: Bitvector[MAX_PERSISTENT_COMMITTEE_SIZE * 2]
```

Comment:
* 2 is because a persistent committee is taken from parts of two adjacent period committees, and in a hypothetical worst case it ends up being twice the size of a period committee.

Reply:
We can do a select from each that prevents ever hitting greater than `MAX_PERSISTENT_COMMITTEE_SIZE`

Reply:
The simplest way to do that would probably be to just do a simple crop (if the committee exceeds that size, chop off the last few members). I could see it being reasonable to do that.

(The approach you suggested would I think break the invariant that a validator cannot be a member of multiple shard committees at the same time)